modules/tinymce/src/plugins/table/test/ts/browser/InsertColumnTableResizeTest.ts

```diff
@@ -1,200 +1,557 @@
-  );
-
-  const cAssertWidth = (label: string, data: TestData) => Chain.label(
-    `Assert width of table ${label} after inserting column`,
-    NamedChain.asChain([
-      NamedChain.direct(NamedChain.inputName(), Chain.identity, 'editor'),
-      NamedChain.direct('editor', cInsertColumnMeasureWidth(label, data), 'widths'),
-      NamedChain.read('widths', cAssertWidths)
-    ])
-  );
-
-  NamedChain.pipeline(Log.chains('TBA', 'Table: Insert columns, erase column and assert the table width does not change', [
-    NamedChain.write('editor', McEditor.cFromSettings({
-      plugins: 'table',
-      width: 400,
-      theme: 'silver',
-      base_url: '/project/tinymce/js/tinymce'
-    })),
-
-    NamedChain.read('editor', cAssertWidth('which is empty', emptyTable)),
-    NamedChain.read('editor', cAssertWidth('with contents in some cells', contentsInSomeCells)),
-    NamedChain.read('editor', cAssertWidth('with contents in all cells', contentsInAllCells)),
-    NamedChain.read('editor', cAssertWidth('with headings', tableWithHeadings)),
-    NamedChain.read('editor', cAssertWidth('with caption', tableWithCaption)),
-    NamedChain.read('editor', cAssertWidth('with nested tables', nestedTables)),
-
-    NamedChain.read('editor', McEditor.cRemove)
-  ]),
-  success, failure, TestLogs.init());
+
+describe('browser.tinymce.plugins.table.InsertColumnTableResizeTest', () => {
+  /**
+   * Test the width of tables before and after column operations
+   * There is a 2 pixel delta allowed from the expected width, to account for table borders
+   */
+
+  before(() => {
+    Plugin();
+  });
+
+  const pCreateEditor = (settings: RawEditorSettings) => McEditor.pFromSettings<Editor>({
+    ...settings,
+    plugins: 'table',
+    base_url: '/project/tinymce/js/tinymce',
+  });
+
+  const getTableWidth = (editor: Editor) =>
+    SelectorFind.child<HTMLTableElement>(TinyDom.body(editor), 'table').map((table) => Width.get(table)).getOrDie();
+
+  // TODO: tests for colspan #TINY-6949
+  context('Responsive table', () => {
+    context('table_column_resizing=preservetable', () => {
+      it('will resize table because responsive tables cannot honour this setting', async () => {
+        const editor = await pCreateEditor({
+          table_sizing_mode: 'responsive',
+          table_column_resizing: 'preservetable',
+        });
+        editor.setContent(`
+          <table>
+            <tbody>
+              <tr>
+                <td></td>
+                <td data-mce-selected="1"></td>
+              </tr>
+              <tr>
+                <td></td>
+                <td></td>
+              </tr>
+            </tbody>
+          </table>
+        `);
+        const beforeWidth = getTableWidth(editor);
+        editor.selection.select(editor.dom.select('td[data-mce-selected]')[0], true);
+        editor.execCommand('mceTableInsertColAfter');
+        const afterWidth = getTableWidth(editor);
+        assert.closeTo(afterWidth, beforeWidth * 1.5, 2);
```

Comment:
Did we end up deciding on `closeTo` vs `approximately`? I certainly prefer the latter, as to me it's clearer it's doing an approximation.

Also, this set selection and do assertion logic seems to be duplicated a lot. Can it be abstracted out to a helper function to remove the duplication?

- [ ] Extract duplication out into a helper function